### PR TITLE
[REF] sale: invoicing wizard

### PR DESCRIPTION
**Pull request by Feyensv**
Enterprise PR: https://github.com/odoo/enterprise/pull/30653
Upgrade PR: https://github.com/odoo/upgrade/pull/3799

Task-2959699

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/sale/wizard/sale_make_invoice_advance_views.xml [hunk 1]
```diff
@@ -1,64 +1,80 @@
...
+                        attrs="{'invisible': ['|', ('has_down_payments', '=', False), ('advance_payment_method', '!=', 'delivered')]}"/>
+                    <div attrs="{'invisible': ['|', ('has_down_payments', '=', False), ('advance_payment_method', '!=', 'delivered')]}"
+                        id="down_payment_details">
+                        <field name="deduct_down_payments" nolabel="1"/>
+                        <label for="deduct_down_payments"/>
+                    </div>
+                </group>
+                <group name="down_payment_specification"
+                    attrs="{'invisible': [('advance_payment_method', 'not in', ('fixed','percentage'))]}">
+                    <field name="product_id" invisible="1"/>
```
Comment (Feyensv) [inline addons/sale/wizard/sale_make_invoice_advance_views.xml:31]: TODO add company_id to trigger product_id computation...

File: addons/sale/wizard/sale_make_invoice_advance.py [hunk 2]
```diff
@@ -5,198 +5,251 @@
...
+                wizard.currency_id = wizard.sale_order_ids.currency_id
+
+    @api.depends('sale_order_ids')
+    def _compute_company_id(self):
+        self.company_id = False
+        for wizard in self:
+            if wizard.count == 1:
+                wizard.company_id = wizard.sale_order_ids.company_id
+
+    @api.depends('company_id')
```
Comment (Feyensv) [inline addons/sale/wizard/sale_make_invoice_advance.py:99]: dumb depends until dp products are specified by company :D

File: addons/sale/wizard/sale_make_invoice_advance.py [hunk 3]
```diff
@@ -5,198 +5,256 @@
...
+                raise UserError(_(
+                    "The product used to invoice a down payment should be of type 'Service'."
+                    " Please use another product or update this product."))
+
+    #=== ACTION METHODS ===#
+
+    def create_invoices(self):
+        self._create_invoices(self.sale_order_ids)
+
+        if self.env.context.get('open_invoices', False):
```
Comment (Demesmaeker) [inline addons/sale/wizard/sale_make_invoice_advance.py:150]: What's the problem of using the default value `None` here?

File: addons/sale/wizard/sale_make_invoice_advance_views.xml [hunk 4]
```diff
@@ -1,64 +1,81 @@
...
+                        string=""
+                        attrs="{'invisible': ['|', ('has_down_payments', '=', False), ('advance_payment_method', '!=', 'delivered')]}"/>
+                    <div attrs="{'invisible': ['|', ('has_down_payments', '=', False), ('advance_payment_method', '!=', 'delivered')]}"
+                        id="down_payment_details">
+                        <field name="deduct_down_payments" nolabel="1"/>
+                        <label for="deduct_down_payments"/>
+                    </div>
+                </group>
+                <group name="down_payment_specification"
+                    attrs="{'invisible': [('advance_payment_method', 'not in', ('fixed','percentage'))]}">
```
Comment (Feyensv) [inline addons/sale/wizard/sale_make_invoice_advance_views.xml:30]:
```suggestion
                    attrs="{'invisible': [('advance_payment_method', 'not in', ('fixed', 'percentage'))]}">
```

**Review by Demesmaeker — COMMENTED** [review]
:duck:

File: addons/sale/wizard/sale_make_invoice_advance.py [hunk 5]
```diff
@@ -5,198 +5,256 @@
...
+        self.company_id = False
+        for wizard in self:
+            if wizard.count == 1:
+                wizard.company_id = wizard.sale_order_ids.company_id
+
+    @api.depends('company_id')  # 'dumb' depends to trigger the computation
+    def _compute_product_id(self):
+        self.product_id = False
+        dp_product_id = int(self.env['ir.config_parameter'].sudo().get_param(
+            'sale.default_deposit_product_id'))
```
[the remark this answers is not in the document]
Comment (Feyensv) [reply addons/sale/wizard/sale_make_invoice_advance.py:105]: yup, but negligible, and it's a better practice to keep this kind of variable fetch outside the loop.

**Comment by Feyensv** [comment]
@robodoo r+ rebase-ff

**Pull request closed without merging**